src/plugins/discover/public/application/main/components/layout/discover_layout.tsx

```diff
@@ -145,6 +147,27 @@ export function DiscoverLayout({ stateContainer }: DiscoverLayoutProps) {
     [filterManager, dataView, dataViews, trackUiMetric, capabilities]
   );
 
+  const onPopulateWhereClause = useCallback(
+    (field: DataViewField | string, values: unknown, operation: '+' | '-') => {
+      if (query && isOfAggregateQueryType(query) && 'esql' in query) {
+        const fieldName = typeof field === 'string' ? field : field.name;
+        // send the field type for casting
+        const fieldType = typeof field !== 'string' ? field.type : undefined;
+        const updatedQuery = appendWhereClauseToESQLQuery(
+          query.esql,
+          fieldName,
+          values,
+          operation,
+          fieldType
+        );
+        data.query.queryString.setQuery({
```

Comment:
Can we also track clicks as in `onAddFilter`?

Reply:
Sure! I will use a different event `esql_filter_added` unless you want to use the same but I feel that a different event will be easier to track, no?

Reply:
https://github.com/elastic/kibana/pull/181399/commits/a6ddb094a5490fe1f2255d6cb4961e0817ace9f5